File: Project3/online-crypto/task2/task2.cpp
#include <iostream>
#include <string>
#include <stdio.h>
using namespace std;

string enum_key(string current) {
    /*Return the next key based on the current key as hex string.

     TODO: Implement the required functions.
     */
	return "Your should implement this function! We are going to test it!";
}

int main(int argc, char* argv[]) {
	string mode = argv[1];
	if (mode == "enum_key") {
		printf("%s", enum_key(argv[2]).c_str());
	} else if (mode == "crack") {
		// TODO: Add your own code and do whatever you do.
	} else {
		printf("Wrong mode!");
	}
	return 0;
}

File: Project3/online-crypto/task2/task2.java
public class task2 {
	static String enumKey(String current) {
        /*Return the next key based on the current key as hex string.
        
        TODO: Implement the required functions.
        */
		return "Your should implement this function! We are going to test it!";
	}

	public static void main (String[] args) {
		String mode = args[0];
		if (mode.equals("enum_key")) {
			System.out.println(enumKey(args[1]));
		} else if (mode.equals("crack")) {
			// TODO: Add your own code and do whatever you do.
		} else {
			System.out.println("Wrong mode!");
		}

	}
}
